src/scripts/index.js

```diff
@@ -0,0 +1,67 @@
+var dayButtons = new Vue({
+    el: "#dayButtons",
+
+    data: {
+        days: []
+    },
+
+    methods: {
+        toHexFromRGB(color) {
+            function convertToHex(c) {
+                let hex = c.toString(16);
+                return hex.length === 1 ? "0" + hex : hex;
+            }
+            return "#" + convertToHex(color[0]) + convertToHex(color[1]) + convertToHex(color[2]);
+        },
+
+        isDayFilled(day) {
+            for(let filledDay of this.days) {
+                if(Date.parse(filledDay.date) === Date.parse(day.toISOString())) {
+                    return true;
+                }
+            }
+            return false;
+        },
+
+        getFilledDayColor(day) {
+            for(let filledDay of this.days) {
+                if(Date.parse(filledDay.date) === Date.parse(day.toISOString())) {
+                    return this.toHexFromRGB(filledDay.average_color);
+                }
+            }
+            return "#a9a9a9";
+        },
+
+        getDisplayedWeeks() {
+            let closestSunday = new Date();
+            closestSunday = new Date(Date.UTC(closestSunday.getFullYear(), closestSunday.getMonth(),
+                closestSunday.getDate() + 7 - closestSunday.getDay()));
+            let threeWeeks = Array.from(Array(3), () => new Array(7));
+            for(let i = 0; i < 3; ++i) {
+                for(let j = 0; j < 7; ++j) {
+                    threeWeeks[i][j] = new Date(closestSunday);
+                    threeWeeks[i][j].setUTCDate(closestSunday.getUTCDate() - 20 + i * 7 + j);
+                }
+            }
+            return threeWeeks;
+        }
+    },
+
+    mounted() {
+        axios.get("/api/days/brief").then((response) => {
+            if (response.data.ok) {
+                this.days = response.data.days;
+                this.days.map((day) => {
+                    day.data = new Date(day.data);
```

Comment:
Field `data` of `day` is a field mentioned in API docs as a field of API responce, when `Date` is a javascript object. If it bothers you, this can be fixed and field `data` can be renamed to `date`, but I think not in this PR. You may open an issue

Reply:
@tanya-kta, I don't understand what you're talking about. I can't find the word "data" in the [`/api/days/brief`'s documentation](https://github.com/LifeCalendarTeam/life_calendar/tree/master/docs/api#apidaysbrief)

Note: such key is mentioned in the docs of `/api/activities` (and it also exists in `/api/emotions`, because these methods have the same interface), but this piece of code doesn't seem to be anyhow related to those methods

Reply:
Yeah, sorry, I wasn't right. That's a bug